### PR TITLE
Identity and attest key triples rewording

### PR DESCRIPTION
**Pull request by nedmsmith**
Issue #126 anticipates better clarification for identity and attest-key triples to disambiguate from keys as measurements.

### CODE REVIEW

File: draft-ietf-rats-corim.md [hunk 1]
```diff
@@ -1054,6 +1054,19 @@ The types or associating addressing information to a measured environment are:
 
 ##### Crypto Keys {#sec-crypto-keys}
 
+A `$crypto-key-type-choice` statement identifies a cryptogrpahic key.
+There are several possible formats.
+The usage context helps determine how Verifiers process each varient.
```
Comment (setrofim) [inline draft-ietf-rats-corim.md:1059]:
```suggestion
The usage context helps determine how Verifiers process each variant.
```

File: draft-ietf-rats-corim.md [hunk 2]
```diff
@@ -1054,6 +1054,19 @@ The types or associating addressing information to a measured environment are:
...
+For example, as a Reference Value, the Verifier applies an Evidence matching algorithm.
+As an identity triple, the Verifier may need to obtain a suitable credential to authenticate the Attester.
+
+Additionally, the deploymnet context may add new processing requirements, such as, proof of possession checking,
```
Comment (setrofim) [inline draft-ietf-rats-corim.md:1063]:
```suggestion
Additionally, the deployment context may add new processing requirements, such as, proof of possession checking,
```

**Review by setrofim — APPROVED** [review]

File: draft-ietf-rats-corim.md [hunk 3]
```diff
@@ -1054,6 +1054,19 @@ The types or associating addressing information to a measured environment are:
 
 ##### Crypto Keys {#sec-crypto-keys}
 
+A `$crypto-key-type-choice` statement identifies a cryptogrpahic key.
```
Comment (yogeshbdeshpande) [inline draft-ietf-rats-corim.md:1057]:
```suggestion
A `$crypto-key-type-choice` statement identifies a cryptographic key.
```

**Review by yogeshbdeshpande — COMMENTED** [review]
reviewing

File: draft-ietf-rats-corim.md [hunk 4]
```diff
@@ -1054,6 +1054,19 @@ The types or associating addressing information to a measured environment are:
 
 ##### Crypto Keys {#sec-crypto-keys}
 
+A `$crypto-key-type-choice` statement identifies a cryptographic key.
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:1057]:
```suggestion
A `$crypto-key-type-choice` identifies a cryptographic key.
```

File: draft-ietf-rats-corim.md [hunk 5]
```diff
@@ -1054,6 +1054,19 @@ The types or associating addressing information to a measured environment are:
...
+The usage context helps determine how Verifiers process each variant.
+For example, as a Reference Value, the Verifier applies an Evidence matching algorithm.
+As an identity triple, the Verifier may need to obtain a suitable credential to authenticate the Attester.
+
+Additionally, the deployment context may add new processing requirements, such as, proof of possession checking,
+credential status checking, or distribution of various processing steps.
+For example, in a split Verifier scenario, a first Verifier may challenge the key to sign a nonce
+then compute a digest of the public key that is forwarded to a second Verifier.
+The second Verifier completes verification by locating the certificate,
+performing certificate path validation, revocation, and trust anchor checks.
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:1068]: I'd drop this para entirely as it introduces too much variability without effectively specifying any particular behaviour.  Either that or leave just the bare minimum, i.e.: _"Additionally, the deployment context may add further processing requirements.  These are outside the scope of the present specification."_

File: draft-ietf-rats-corim.md [hunk 6]
```diff
@@ -1128,10 +1136,12 @@ the object relates to the subject.
 #### Device Identity Triple {#sec-comid-triple-identity}
 
 A Device Identity triple relates one or more cryptographic keys to a device.
-The subject of an Identity triple uses an instance or class identifier to refer
+The subject of an identity triple uses an instance or class identifier to refer
 to a device, and a cryptographic key is the object. The predicate asserts that
-the identity is authenticated by the key. A common application for this triple
-is device identity.
+at least one of the keys, as defined by `$crypto-key-type-choice`, authenticates the Attester.
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:1141]: > at least one of the keys, as defined by `$crypto-key-type-choice`, authenticates the Attester

Should it be renamed to "Attester Identity Triple" then?

File: draft-ietf-rats-corim.md [hunk 7]
```diff
@@ -1123,26 +1123,38 @@ the object relates to the subject.
...
-The subject of an identity triple uses an instance or class identifier to refer
-to a device, and a cryptographic key is the object. The predicate asserts that
-at least one of the keys, as defined by `$crypto-key-type-choice`, authenticates the Attester.
+The triple subject is an identity used to name a device.
+The triple object is an array of cryptographic key(s) used to authenticate the device.
+The predicate asserts that any of the key(s) (found in the triple object) can be used to authenticate the device.
+
+If contained in a CoRIM, this triple endorses that the key(s) were provisioned to the Target Environment.
+
+If contained in Evidence, this triple asserts that the key(s) are present in the Target Environment.
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:1132]: This sentence should be in the Concise Evidence spec rather than here.

File: draft-ietf-rats-corim.md [hunk 8]
```diff
@@ -1123,26 +1123,38 @@ the object relates to the subject.
...
-Environment whose object is a cryptographic key. The predicate asserts that the
-Attesting Environment integrity protects Evidence using at least one of the keys.
+An Attestation Keys triple relates one or more cryptographic keys to an Attesting Environment.
+The attest key triple subject identifies an Attesting Environment
+whose object is an array of cryptographic key(s) used to attest Evidence.
+The predicate asserts that any of the key(s) (found in the triple object) can be used to attest Evidence.
+
+If contained in a CoRIM, this triple endorses that the key(s) were provisioned to the Attesting Environment.
+
+If contained in Evidence, this triple asserts that the key(s) are present in the Attesting Environment.
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:1152]: This sentence should be in the Concise Evidence spec rather than here.

File: draft-ietf-rats-corim.md [hunk 9]
```diff
@@ -1123,26 +1123,38 @@ the object relates to the subject.
 #### Device Identity Triple {#sec-comid-triple-identity}
 
 A Device Identity triple relates one or more cryptographic keys to a device.
-The subject of an identity triple uses an instance or class identifier to refer
-to a device, and a cryptographic key is the object. The predicate asserts that
-at least one of the keys, as defined by `$crypto-key-type-choice`, authenticates the Attester.
+The triple subject is an identity used to name a device.
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:1126]: Just a nit: maybe we could add a sentence to clarify the distinction between DI and AK triples?

**Review by thomas-fossati — COMMENTED** [review]
LGTM, thanks -- with a couple of sentences removed and a possible one-line clarification added (see inline)

File: draft-ietf-rats-corim.md [hunk 10]
```diff
@@ -1128,21 +1123,38 @@ the object relates to the subject.
...
-is device identity.
+The triple subject is an identity used to name a device.
+The triple object is an array of cryptographic key(s) used to authenticate the device.
+The predicate asserts that any of the key(s) (found in the triple object) can be used to authenticate the device.
+
+If contained in a CoRIM, this triple endorses that the key(s) were provisioned to the Target Environment.
+
+If contained in Evidence, this triple asserts that the key(s) are present in the Target Environment.
+
+The Verifier SHOULD attempt to verify the status and validity of the key(s).
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1134]:
```suggestion
The Verifier SHOULD perform offline verification of the key(s).
```

File: draft-ietf-rats-corim.md [hunk 11]
```diff
@@ -1128,21 +1123,38 @@ the object relates to the subject.
...
+An Attestation Keys triple relates one or more cryptographic keys to an Attesting Environment.
+The attest key triple subject identifies an Attesting Environment
+whose object is an array of cryptographic key(s) used to attest Evidence.
+The predicate asserts that any of the key(s) (found in the triple object) can be used to attest Evidence.
+
+If contained in a CoRIM, this triple endorses that the key(s) were provisioned to the Attesting Environment.
+
+If contained in Evidence, this triple asserts that the key(s) are present in the Attesting Environment.
+
+The Verifier SHOULD attempt to verify the status and validity of the key(s).
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1154]:
```suggestion
The Verifier SHOULD perform offline verification of the key(s).
```

**Review by nedmsmith — COMMENTED** [review]
tweaked wording to include offline key verification

File: draft-ietf-rats-corim.md (hunk 8 above)
Comment (nedmsmith) [reply draft-ietf-rats-corim.md:1152]: > This sentence should be in the Concise Evidence spec rather than here.

When a claim is added to the ACS, the semantics of the claim are asserted by the verifier. This specification should capture these semantics. If another spec defines evidence that is mapped into this schema. The mapping expects the semantics are the same since mapping different semantics can be challenging. Stating it here ensures the other evidence forms are semantically similar/same.

File: draft-ietf-rats-corim.md [hunk 12]
```diff
@@ -1128,21 +1123,38 @@ the object relates to the subject.
...
+
+Depending on which `$crypto-key-type-choice` form is supplied, the Verifier may
+need to take different steps to locate or verify the key.
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1158]:
```suggestion
Note: some keys can only be used for attestation purposes, hence should not be mistaken for a device identities.
```

File: draft-ietf-rats-corim.md (hunk 9 above)
Comment (nedmsmith) [reply draft-ietf-rats-corim.md:1126]: > Just a nit: maybe we could add a sentence to clarify the distinction between DI and AK triples?

see suggested text at line 1158

File: draft-ietf-rats-corim.md [hunk 13]
```diff
@@ -1128,21 +1123,38 @@ the object relates to the subject.
...
 
 #### Attestation Keys Triple {#sec-comid-triple-attest-key}
 
-An Attestation Keys triple relates one or more cryptographic keys to an
-Attesting Environment. The Attestation Key triple subject is an Attesting
-Environment whose object is a cryptographic key. The predicate asserts that the
-Attesting Environment signs Evidence that can be verified using the key.
+An Attestation Keys triple relates one or more cryptographic keys to an Attesting Environment.
+The attest key triple subject identifies an Attesting Environment
+whose object is an array of cryptographic key(s) used to attest Evidence.
```
Comment (yogeshbdeshpande) [inline draft-ietf-rats-corim.md:1147]: attest the Evidence on line 1147 is confusing and not really ties to any Cryptographic operation.

Comment (yogeshbdeshpande) [inline draft-ietf-rats-corim.md:1147]:
```suggestion
The attest key triple subject identifies an Attesting Environment whose object
is an array of of cryptographic keys used to validate the Integrity of Evidence.
```

File: draft-ietf-rats-corim.md [hunk 14]
```diff
@@ -1128,21 +1123,38 @@ the object relates to the subject.
...
+The predicate asserts that any of the key(s) (found in the triple object) can be used to attest Evidence.
```
Comment (yogeshbdeshpande) [inline draft-ietf-rats-corim.md:1148]:
```suggestion
The predicate asserts that any of the key(s) (found in the triple object) can be used to Validate integrity of Evidence.
```

File: draft-ietf-rats-corim.md [hunk 15]
```diff
@@ -1472,9 +1484,176 @@ is extended using Endorsements etc. from the accepted tags.
 
 ### Matching Evidence against Reference Values
 
+An Endorser may use CoMID tags to publish Conditional Endorsements, which
+are added to the Accepted Claims Set only if specified conditions apply.
+This section describes the process performed by the Verifier to determine
+which Conditional Endorsements from the candidate CoMIDs should be added
+to the Accepted Claims Set.
+
```
Comment (yogeshbdeshpande) [inline draft-ietf-rats-corim.md:1492]: If I am not mistaken, I guess this section is already merged ?

File: draft-ietf-rats-corim.md (hunk 13 above)
Comment (nedmsmith) [inline draft-ietf-rats-corim.md]:
```suggestion
whose object is an array of cryptographic key(s) used to verify the integrity of Evidence.
```

**Review by nedmsmith — COMMENTED** [review]
accepted suggested text

**Comment by nedmsmith** [comment]
Updated issue #126 with an alternative approach than this PR.

File: draft-ietf-rats-corim.md [hunk 16]
```diff
@@ -1138,21 +1133,47 @@ the object relates to the subject.
...
 
 A Device Identity triple relates one or more cryptographic keys to a device.
-The subject of an Identity triple uses an instance or class identifier to refer
+The triple subject is an identity used to name a device.
+The triple object is an array of cryptographic key(s) used to authenticate the device.
+The predicate asserts that any of the key(s) (found in the triple object) can be used to authenticate the device.
+
+If contained in a CoRIM, this triple endorses that the key(s) were provisioned to the Target Environment.
+
+If contained in Evidence, this triple asserts that the key(s) are present in the Target Environment.
```
Comment (henkbirkholz) [inline draft-ietf-rats-corim.md:1142]:
```suggestion
```

File: draft-ietf-rats-corim.md [hunk 17]
```diff
@@ -1138,21 +1133,45 @@ the object relates to the subject.
...
+An Attestation Keys triple relates one or more cryptographic keys to an Attesting Environment.
+The attest key triple subject identifies an Attesting Environment whose object
+is an array of cryptographic keys used to validate the integrity of Evidence.
+The predicate asserts that any of the key(s) (found in the triple object) can be used to validate integrity of Evidence.
+
+If contained in a CoRIM, this triple endorses that the key(s) were provisioned to the Attesting Environment.
+
+If contained in Evidence, this triple asserts that the key(s) are present in the Attesting Environment.
+
+The Verifier SHOULD perform offline verification of the key(s).
```
Comment (yogeshbdeshpande) [inline draft-ietf-rats-corim.md:1166]: The statement seems ambiguous to me? Can you please clarify what does it means? Did you mean CRL checking or something ?

File: draft-ietf-rats-corim.md [hunk 18]
```diff
@@ -1138,21 +1133,45 @@ the object relates to the subject.
...
+
 An Attestation Keys triple relates one or more cryptographic keys to an
-Attesting Environment. The Attestation Key triple subject is an Attesting
+Attesting Environment. The attest key triple subject is an Attesting
```
Comment (yogeshbdeshpande) [inline draft-ietf-rats-corim.md:1169]:
```suggestion
Attesting Environment. The attest key triple subject is an Attesting
Environment whose object is one or more cryptographic keys.
```

File: draft-ietf-rats-corim.md (hunk 6 above)
Comment (yogeshbdeshpande) [reply draft-ietf-rats-corim.md:1146]: Here device is the application layer concept that is been mapped to a `Environment` for encoding.

Perhaps, we can clarify better that the Device Identifier, either class of Device or specific Device instance, is expressed in the Environment in the subject of the triple

**Review by yogeshbdeshpande — COMMENTED** [review]
In general LGTM, Some tidy up is required but this PR is very much relevant to define the semantics of the two triples which needs better explanation than the existing text

File: draft-ietf-rats-corim.md [hunk 19]
```diff
@@ -1138,22 +1133,46 @@ the object relates to the subject.
...
+
 An Attestation Keys triple relates one or more cryptographic keys to an
-Attesting Environment. The Attestation Key triple subject is an Attesting
-Environment whose object is a cryptographic key. The predicate asserts that the
-Attesting Environment signs Evidence that can be verified using the key.
+Attesting Environment. The attest key triple subject is an Attesting
+Environment whose object is one or more cryptographic keys.
+The predicate asserts that the Attesting Environment integrity protects Evidence using at least one of the keys.
 
+Depending on which `$crypto-key-type-choice` form is supplied, the Verifier may
```
Comment (deeglaze) [inline draft-ietf-rats-corim.md:1173]: Remove line break without new sentence

File: draft-ietf-rats-corim.md [hunk 20]
```diff
@@ -1138,22 +1133,46 @@ the object relates to the subject.
...
 
 ~~~ cddl
 {::include cddl/identity-triple-record.cddl}
 ~~~
 
 #### Attestation Keys Triple {#sec-comid-triple-attest-key}
 
+An Attestation Keys triple relates one or more cryptographic keys to an Attesting Environment.
+The attest key triple subject identifies an Attesting Environment whose object
+is an array of cryptographic keys used to validate the integrity of Evidence.
```
Comment (deeglaze) [inline draft-ietf-rats-corim.md:1159]: Remove line break without new sentence

File: draft-ietf-rats-corim.md [hunk 21]
```diff
@@ -1138,22 +1133,46 @@ the object relates to the subject.
...
+The Verifier SHOULD perform offline verification of the key(s).
+
+The subject of an identity triple uses an instance or class identifier to refer
 to a device, and a cryptographic key is the object. The predicate asserts that
-the identity is authenticated by the key. A common application for this triple
-is device identity.
+at least one of the keys, as defined by `$crypto-key-type-choice`, authenticates the Attester.
+
+Depending on which `$crypto-key-type-choice` form is supplied, the Verifier may
+need to take different steps to locate or verify the key.
```
Comment (deeglaze) [inline draft-ietf-rats-corim.md:1149]: Remove line break without new sentence.

File: draft-ietf-rats-corim.md [hunk 22]
```diff
@@ -1138,22 +1133,46 @@ the object relates to the subject.
...
+
+If contained in a CoRIM, this triple endorses that the key(s) were provisioned to the Target Environment.
+
+The Verifier SHOULD perform offline verification of the key(s).
+
+The subject of an identity triple uses an instance or class identifier to refer
 to a device, and a cryptographic key is the object. The predicate asserts that
-the identity is authenticated by the key. A common application for this triple
-is device identity.
+at least one of the keys, as defined by `$crypto-key-type-choice`, authenticates the Attester.
```
Comment (deeglaze) [inline draft-ietf-rats-corim.md:1146]: Remove line break without new sentence.

File: draft-ietf-rats-corim.md [hunk 23]
```diff
@@ -1138,22 +1133,46 @@ the object relates to the subject.
...
 
 A Device Identity triple relates one or more cryptographic keys to a device.
-The subject of an Identity triple uses an instance or class identifier to refer
+The triple subject is an identity used to name a device.
+The triple object is an array of cryptographic key(s) used to authenticate the device.
+The predicate asserts that any of the key(s) (found in the triple object) can be used to authenticate the device.
+
+If contained in a CoRIM, this triple endorses that the key(s) were provisioned to the Target Environment.
+
+The Verifier SHOULD perform offline verification of the key(s).
```
Comment (deeglaze) [inline draft-ietf-rats-corim.md:1142]: What is the significance of the "offline" qualification?

**Review by deeglaze — CHANGES_REQUESTED** [review]

**Comment by yogeshbdeshpande** [comment]
Ned to progress this on return!

File: draft-ietf-rats-corim.md [hunk 24]
```diff
@@ -1138,22 +1133,46 @@ the object relates to the subject.
...
 #### Attestation Keys Triple {#sec-comid-triple-attest-key}
 
+An Attestation Keys triple relates one or more cryptographic keys to an Attesting Environment.
+The attest key triple subject identifies an Attesting Environment whose object
+is an array of cryptographic keys used to validate the integrity of Evidence.
+The predicate asserts that any of the key(s) (found in the triple object) can be used to validate integrity of Evidence.
+
+If contained in a CoRIM, this triple endorses that the key(s) were provisioned to the Attesting Environment.
+
+If contained in Evidence, this triple asserts that the key(s) are present in the Attesting Environment.
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:1164]:
```suggestion
```

File: draft-ietf-rats-corim.md (hunk 6 above)
Comment (thomas-fossati) [reply draft-ietf-rats-corim.md:1146]: @nedmsmith to wordsmith this

**Comment by nedmsmith** [comment]
I resolved most of the conversations due to outdated content. Revised text is provided in the most recent updates.

File: draft-ietf-rats-corim.md [hunk 25]
```diff
@@ -1124,22 +1119,46 @@ the object relates to the subject.
...
-is device identity.
+A Device Identity triple record relates one or more cryptographic keys to a device identity.
+The cryptographic keys are bound to or associated with a Target Environment that is within the device.
+The device identifier may be part of the Target Environment's `environment-map` or may be part of some other device identity credential, such as a certificate.
+The cryptographic keys are expected to be used to authenticate the device.
+
+Device Identity triples instruct a Verifier to perform key validation checks, such as revocation, certificate path construction & verification, or proof of possession.
+The Verifier SHOULD perform offline verification of keys contained in Device Identity triples.
+
+When Device Identity triples are contained in a CoRIM, it endorses that the keys were securely provisioned to the named Target Environment.
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:1130]:
```suggestion
A Device Identity triple endorses that the keys were securely provisioned to the named Target Environment.
```

File: draft-ietf-rats-corim.md [hunk 26]
```diff
@@ -1124,22 +1119,46 @@ the object relates to the subject.
...
+Additional details about how a key was provisioned or is protected may be asserted using Endorsements such as  `endorsed-triples`.
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:1131]: Should we say how? Or, if we do already, reference the place(s) that contain the relevant text.

File: draft-ietf-rats-corim.md [hunk 27]
```diff
@@ -1124,22 +1119,46 @@ the object relates to the subject.
...
+
+Depending on key formatting, as defined by `$crypto-key-type-choice`, the Verifier may take different steps to locate and verify the key.
+
+If a key has usage restrictions that limits its use to device identity challenges, Verifiers SHOULD check for key use that violates usage restrictions.
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:1135]:
```suggestion
If a key has usage restrictions that limit its use to device identity challenges, Verifiers SHOULD check for key use that violates usage restrictions.
```

File: draft-ietf-rats-corim.md [hunk 28]
```diff
@@ -1124,22 +1119,46 @@ the object relates to the subject.
...
 
-#### Attestation Keys Triple {#sec-comid-triple-attest-key}
+#### Attest Key Triple {#sec-comid-triple-attest-key}
+
+An Attest Key triple record relates one or more cryptographic keys to an Attesting Environment.
+The cryptographic keys are wielded by an Attesting Environment that collects measurements from a Target Environment.
+The cryptographic keys sign Evidence.
+
+Attest Key triples instruct a Verifier to perform key validation checks, such as revocation, certificate path construction & verification, or proof of possession.
+The Verifier SHOULD perform offline verification of keys contained in Attest Key triples.
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:1151]: Isn't this a MUST?  The overall security of the system depends on being able to validate evidence's signatures.

File: draft-ietf-rats-corim.md [hunk 29]
```diff
@@ -1124,22 +1119,46 @@ the object relates to the subject.
...
+
+When Attest Key triples are contained in a CoRIM, it endorses that the keys were securely provisioned to the named Attesting Environment.
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:1153]:
```suggestion
Attest Key triples endorse that the keys were securely provisioned to the named Attesting Environment.
```

File: draft-ietf-rats-corim.md [hunk 30]
```diff
@@ -1124,22 +1119,46 @@ the object relates to the subject.
...
+Additional details about how a key was provisioned or is protected may be asserted using Endorsements such as  `endorsed-triples`.
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:1154]: ditto (symmetric to identity keys)

File: draft-ietf-rats-corim.md [hunk 31]
```diff
@@ -1124,22 +1119,46 @@ the object relates to the subject.
...
+
+Depending on key formatting, as defined by `$crypto-key-type-choice`, the Verifier may take different steps to locate and verify the key.
+
+If a key has usage restrictions that limits its use to Evidence signing, Verifiers SHOULD check for key use that violates usage restrictions.
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:1158]: To me this too is a MUST.

**Review by yogeshbdeshpande — APPROVED** [review]
LGTM!

File: draft-ietf-rats-corim.md (hunk 28 above)
Comment (nedmsmith) [reply draft-ietf-rats-corim.md:1151]: If the verifier decides not to do the check and therefore doesn't add a claim containing the verification results claim and the RP expects to find the claim, then the RP applies the enforcement decision and likely doesn't grant access to the resource requested by the Attester. This approach would not force the verifier into becoming an enforcement point if a deployment architecture didn't want that.

File: draft-ietf-rats-corim.md (hunk 30 above)
Comment (nedmsmith) [reply draft-ietf-rats-corim.md:1154]: I'm confused what action to take?

File: draft-ietf-rats-corim.md (hunk 31 above)
Comment (nedmsmith) [reply draft-ietf-rats-corim.md:1158]: See answer to the previous question about MUST

File: draft-ietf-rats-corim.md (hunk 28 above)
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md:1151]: I am not convinced that your argument justifies a SHOULD, but for the sake of the cut-off, this is good enough for now.

File: draft-ietf-rats-corim.md (hunk 26 above)
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md:1131]: Let's move that issue to -08

File: draft-ietf-rats-corim.md [hunk 32]
```diff
@@ -1124,22 +1119,46 @@ the object relates to the subject.
...
 {::include cddl/identity-triple-record.cddl}
 ~~~
 
-#### Attestation Keys Triple {#sec-comid-triple-attest-key}
+#### Attest Key Triple {#sec-comid-triple-attest-key}
+
+An Attest Key triple record relates one or more cryptographic keys to an Attesting Environment.
+The cryptographic keys are wielded by an Attesting Environment that collects measurements from a Target Environment.
+The cryptographic keys sign Evidence.
+
```
Comment (henkbirkholz) [inline draft-ietf-rats-corim.md:1149]:
```suggestion
```
Co-authored-by: Ned Smith <nedmsmith@users.noreply.github.com>
nedmsmith

File: draft-ietf-rats-corim.md [hunk 33]
```diff
@@ -1124,22 +1119,45 @@ the object relates to the subject.
...
-#### Attestation Keys Triple {#sec-comid-triple-attest-key}
+#### Attest Key Triple {#sec-comid-triple-attest-key}
+
+An Attest Key triple record relates one or more cryptographic keys to an Attesting Environment.
+The cryptographic keys are wielded by an Attesting Environment that collects measurements from a Target Environment.
+The cryptographic keys sign Evidence.
+Attest Key triples instruct a Verifier to perform key validation checks, such as revocation, certificate path construction & verification, or proof of possession.
+The Verifier SHOULD perform offline verification of keys contained in Attest Key triples.
+
+Attest Key triples endorse that the keys were securely provisioned to the named Attesting Environment.
```
Comment (henkbirkholz) [inline draft-ietf-rats-corim.md:1152]:
```suggestion
Attest Key triples endorse that the keys were securely provisioned to the named (identified via an Environment Map) Attesting Environment.
```

File: draft-ietf-rats-corim.md (hunk 30 above)
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md:1153]: If there should be additional symmetric text about identity key, that goes into -08

File: draft-ietf-rats-corim.md [hunk 34]
```diff
@@ -1124,22 +1119,45 @@ the object relates to the subject.
...
+The cryptographic keys are wielded by an Attesting Environment that collects measurements from a Target Environment.
+The cryptographic keys sign Evidence.
+Attest Key triples instruct a Verifier to perform key validation checks, such as revocation, certificate path construction & verification, or proof of possession.
+The Verifier SHOULD perform offline verification of keys contained in Attest Key triples.
+
+Attest Key triples endorse that the keys were securely provisioned to the named (identified via an Environment Map) Attesting Environment.
+Additional details about how a key was provisioned or is protected may be asserted using Endorsements such as  `endorsed-triples`.
+
+Depending on key formatting, as defined by `$crypto-key-type-choice`, the Verifier may take different steps to locate and verify the key.
+
```
Comment (henkbirkholz) [inline draft-ietf-rats-corim.md:1156]:
```suggestion
```

**Review by henkbirkholz — APPROVED** [review]
let's :ship: this for now

File: draft-ietf-rats-corim.md [hunk 35]
```diff
@@ -1120,22 +1115,44 @@ the object relates to the subject.
...
+Device Identity triples instruct a Verifier to perform key validation checks, such as revocation, certificate path construction & verification, or proof of possession.
+The Verifier SHOULD perform offline verification of keys contained in Device Identity triples.
+
+A Device Identity triple endorses that the keys were securely provisioned to the named Target Environment.
+Additional details about how a key was provisioned or is protected may be asserted using Endorsements such as  `endorsed-triples`.
+
+Depending on key formatting, as defined by `$crypto-key-type-choice`, the Verifier may take different steps to locate and verify the key.
+
+If a key has usage restrictions that limit its use to device identity challenges, Verifiers SHOULD check for key use that violates usage restrictions.
```
Comment (deeglaze) [inline draft-ietf-rats-corim.md:1131]: Is this the only key use limitation that should imply key usage checks? What key usage is associated with device identity challenges? Non-repudiation? I'm only considering x.509 key usage OIDs here. What else could you be referring to?

File: draft-ietf-rats-corim.md [hunk 36]
```diff
@@ -1120,22 +1115,44 @@ the object relates to the subject.
...
+
+Offline verification of keys or verification of key use restrictions MAY produce Claims that are added to the ACS.
```
Comment (deeglaze) [inline draft-ietf-rats-corim.md:1133]: We already know that a Verifier can add claims of whatever sort. What measurement-value-map code point are you talking about for these? If there is nothing specific, I think this is too vague to be a useful addition.

File: draft-ietf-rats-corim.md [hunk 37]
```diff
@@ -1120,22 +1115,44 @@ the object relates to the subject.
...
-#### Attestation Keys Triple {#sec-comid-triple-attest-key}
+#### Attest Key Triple {#sec-comid-triple-attest-key}
+
+An Attest Key triple record relates one or more cryptographic keys to an Attesting Environment.
+The cryptographic keys are wielded by an Attesting Environment that collects measurements from a Target Environment.
+The cryptographic keys sign Evidence.
+Attest Key triples instruct a Verifier to perform key validation checks, such as revocation, certificate path construction & verification, or proof of possession.
+The Verifier SHOULD perform offline verification of keys contained in Attest Key triples.
+
+Attest Key triples endorse that the keys were securely provisioned to the named (identified via an Environment Map) Attesting Environment.
```
Comment (deeglaze) [inline draft-ietf-rats-corim.md:1148]:
```suggestion
Attest Key triples endorse that the keys were securely provisioned to the named (identified via an `environment-map`) Attesting Environment.
```

The type is used throughout the document. I don't see capitalized usage.

File: draft-ietf-rats-corim.md [hunk 38]
```diff
@@ -1120,22 +1115,44 @@ the object relates to the subject.
...
+Additional details about how a key was provisioned or is protected may be asserted using Endorsements such as  `endorsed-triples`.
```
Comment (deeglaze) [inline draft-ietf-rats-corim.md:1149]:
```suggestion
Additional details about how a key was provisioned or is protected may be asserted using Endorsements such as `endorsed-triples`.
```

File: draft-ietf-rats-corim.md [hunk 39]
```diff
@@ -1120,22 +1115,44 @@ the object relates to the subject.
...
+A Device Identity triple record relates one or more cryptographic keys to a device identity.
+The cryptographic keys are bound to or associated with a Target Environment that is within the device.
+The device identifier may be part of the Target Environment's `environment-map` or may be part of some other device identity credential, such as a certificate.
+The cryptographic keys are expected to be used to authenticate the device.
+
+Device Identity triples instruct a Verifier to perform key validation checks, such as revocation, certificate path construction & verification, or proof of possession.
+The Verifier SHOULD perform offline verification of keys contained in Device Identity triples.
+
+A Device Identity triple endorses that the keys were securely provisioned to the named Target Environment.
+Additional details about how a key was provisioned or is protected may be asserted using Endorsements such as  `endorsed-triples`.
```
Comment (deeglaze) [inline draft-ietf-rats-corim.md:1127]:
```suggestion
Additional details about how a key was provisioned or is protected may be asserted using Endorsements such as `endorsed-triples`.
```

**Review by deeglaze — APPROVED** [review]

File: draft-ietf-rats-corim.md (hunk 35 above)
Comment (nedmsmith) [reply draft-ietf-rats-corim.md:1131]: DICE and SPDM have additional key usage constraints.
